app/commands/v2/unpublish.rb

```diff
@@ -17,7 +22,7 @@ def call
 
         check_version_and_raise_if_conflicting(content_item, previous_version_number)
 
-        if draft_present?(content_id)
+        if draft_present?(content_id) && !payload[:allow_draft]
```

Comment:
I've had a look, but I'm unsure how to make this better. I'm not sure extracting it out in to a method would help, as that is just moving the complexity.